libvirt/tunnel/tunnel.sh

```diff
@@ -1,98 +1,128 @@
+CLUSTER_ID=$(yq eval '.profile.cluster_id' ${filename})
+ENVIRONMENT=$(yq eval '.profile.environment' ${filename})
+TOKEN=$(yq eval '.profile.token' ${filename})
 
 # Debug and verify input
-if [[ -z "${token:-}" ]]; then
-  echo "[ERROR] \$token must be passed."
-  exit 1
+if [[ -z "${TOKEN:-}" ]]; then
+	echo "[FATAL] \${token} must be set to the credentials of the port-forwarder service account."
+	exit 1
+elif [[ -z "${ENVIRONMENT:-}" ]]; then
+	echo "[FATAL] \${environment} must be set to specify which bastion to interact with."
+	exit 1
+elif [[ -z "${CLUSTER_CAPACITY:-}" ]]; then
+	echo "[FATAL] \${CLUSTER_CAPACITY} must be set to specify cluster capacity."
+	exit 1
+elif [[ -z "${CLUSTER_ID:-}" ]]; then
+	echo "[FATAL] \${CLUSTER_ID} must be set to specify cluster."
+	exit 1
 fi
-# declare -p BASTION_PORTS
-# declare -p CLUSTER_PORTS
-# echo "-R ${BASTION_PORTS['LIBVIRT']}:127.0.0.1:16509"
-# echo "-R ${CLUSTER_PORTS['0,API']}:127.0.0.1:6443   -R ${CLUSTER_PORTS['0,HTTP']}:127.0.0.1:80    -R ${CLUSTER_PORTS['0,HTTPS']}:127.0.0.1:443"
-# echo "-R ${CLUSTER_PORTS['1,API']}:127.0.0.1:16443  -R ${CLUSTER_PORTS['1,HTTP']}:127.0.0.1:10080 -R ${CLUSTER_PORTS['1,HTTPS']}:127.0.0.1:10443"
-# echo "-R ${CLUSTER_PORTS['2,API']}:127.0.0.1:26443  -R ${CLUSTER_PORTS['2,HTTP']}:127.0.0.1:20080 -R ${CLUSTER_PORTS['2,HTTPS']}:127.0.0.1:20443"
-# echo "-R ${CLUSTER_PORTS['3,API']}:127.0.0.1:36443  -R ${CLUSTER_PORTS['3,HTTP']}:127.0.0.1:30080 -R ${CLUSTER_PORTS['3,HTTPS']}:127.0.0.1:30443"
-# echo "-R ${CLUSTER_PORTS['4,API']}:127.0.0.1:46443  -R ${CLUSTER_PORTS['4,HTTP']}:127.0.0.1:40080 -R ${CLUSTER_PORTS['4,HTTPS']}:127.0.0.1:40443"
-
-function OC() {
-	./oc --server https://api.ci.openshift.org --token "${token}" --namespace "${bastion_name}" "${@}"
+
+# Declaring and setting Bastion and Local ports
+PORTS="-R $(yq eval '.libvirt.bastion-port' ${filename}):127.0.0.1:$(yq eval '.libvirt.target-port' ${filename})"
+for i in $(seq 0 $(( $CLUSTER_CAPACITY-1 )) ); do
+		PORTS+=" -R $(yq eval '.libvirt-'$ARCH-$CLUSTER_ID-$i'.api.bastion-port' ${filename}):127.0.0.1:$(yq eval '.libvirt-'$ARCH-$CLUSTER_ID-$i'.api.target-port' ${filename}) 
+				 -R $(yq eval '.libvirt-'$ARCH-$CLUSTER_ID-$i'.http.bastion-port' ${filename}):127.0.0.1:$(yq eval '.libvirt-'$ARCH-$CLUSTER_ID-$i'.http.target-port' ${filename}) 
+				 -R $(yq eval '.libvirt-'$ARCH-$CLUSTER_ID-$i'.https.bastion-port' ${filename}):127.0.0.1:$(yq eval '.libvirt-'$ARCH-$CLUSTER_ID-$i'.https.target-port' ${filename}) "
+done
+# echo ${PORTS}
+
+function OC() {	
+	./oc --server https://api.ci.openshift.org --token "${TOKEN}" --namespace "${ENVIRONMENT}" "${@}"
+}
+
+function timestamp() {
+	# With UTC format.  2020-11-04 06:19:24
+	date -u  +"%Y-%m-%d %H:%M:%S"
 }
 
 function port-forward() {
 	while true; do
-		echo "[INFO] Setting up port-forwarding..."
+		echo "$(timestamp) [INFO] Setting up port-forwarding to connect to the bastion..."
```

Comment:
I think we should move this out of the while true; do loop. Otherwise it will consistently echo on each loop.

Reply:
yes, a readme is in progress.